grammar.js

```diff
@@ -2300,6 +2327,44 @@ module.exports = grammar({
       ))
     },
 
+    pseudo_call: $ => {
+      const receiver = seq(field('receiver', $._expression), '.')
+
+      return choice(
+        seq(
+          optional(receiver),
+          field('method', alias(choice('as', 'as?', 'is_a?'), $.identifier)),
+          field('arguments', alias($.pseudo_call_argument_list, $.argument_list)),
+        ),
+        seq(
+          optional(receiver),
+          field('method', alias('nil?', $.identifier)),
+          optional(seq('(', ')')),
+        ),
+        seq(
+          optional(receiver),
+          field('method', alias('responds_to?', $.identifier)),
+          field('arguments', alias($.pseudo_responds_to_argument_list, $.argument_list)),
+        ),
+      )
```

Comment:
I think this can be dropped -- these already parse as regular methods

Reply:
Fair and I'm fine either way, but `.responds_to?` can only take a symbol. May help error handling for tree sitter to know that.

Reply:
(same with `.nil?` not having any args, just optional parens)

Reply:
Okay, doesn't hurt to leave in then